Examples/graph-toy.lisp

```diff
@@ -99,21 +99,22 @@
   ((val :initarg :val :accessor val)))
 
 (defmethod handle-event ((frame graph-toy) (event new-value-event))
-  (vector-push-extend (val event) (val-array frame))
-  (let ((val-array-len (length (val-array frame)))
-        (xval-count (max-xvals frame)))
-    (when (>= val-array-len (* xval-count 2))
-      (loop for i from 0 to (- xval-count 1)
-            for j from (- val-array-len xval-count)
-            do (setf (elt (val-array frame) i) (elt (val-array frame) j)))
-      (setf (fill-pointer (val-array frame)) xval-count))
-    (redisplay-frame-pane frame 'main-display)
-    (incf (last-yval frame) (- (random 9) 4))
-    (when (member (frame-state frame) '(:enabled :shrunk))
-      (schedule-event (frame-top-level-sheet frame)
-                      (make-instance 'new-value-event :sheet frame
-                                                      :val (last-yval frame))
-                      0.5))))
+  (when (eq (frame-state frame) :enabled)
+    (vector-push-extend (val event) (val-array frame))
+    (let ((val-array-len (length (val-array frame)))
+          (xval-count (max-xvals frame)))
+      (when (>= val-array-len (* xval-count 2))
+        (loop for i from 0 to (- xval-count 1)
+           for j from (- val-array-len xval-count)
+           do (setf (elt (val-array frame) i) (elt (val-array frame) j)))
+        (setf (fill-pointer (val-array frame)) xval-count))
+      (redisplay-frame-pane frame 'main-display)
+      (incf (last-yval frame) (- (random 9) 4))
+      (when (member (frame-state frame) '(:enabled :shrunk))
```

Comment:
I guess this answers the above question. But if we already doing this check at the start, do we really need to check again here?